### PR TITLE
Add several new methods to ShadowPaint

### LINKED ISSUE #7246: Several methods from Paint.class aren't implemented in ShadowPaint
### Description
Some methods in Paint.class that makes use of private native methods such as the getters and setters of letterSpacing, textScaleX, textSkewX, wordSpacing... are not implemented in ShadowPaint meaning these does not work in robolectric

### Steps to Reproduce
Having the following matcher:
```
class LetterSpacingMatcher(@DimenRes private val expectedLetterSpacing: Int) :
  TypeSafeMatcher<View>(TextView::class.java) {
  companion object {
    fun hasLetterSpacing(@DimenRes expectedDimenRes: Int): LetterSpacingMatcher =
      LetterSpacingMatcher(expectedDimenRes)
  }

  private var expectedSize: Float = 0F
  private var actualLetterSpacing: Float? = null
  override fun matchesSafely(item: View): Boolean {
    item as TextView
    actualLetterSpacing = item.letterSpacing
    expectedSize = ResourcesCompat.getFloat(item.resources, expectedLetterSpacing);
    return actualLetterSpacing == expectedSize
  }

  override fun describeTo(description: Description) {
    description.appendText("Text did not match line height $expectedSize, was $actualLetterSpacing")
  }
}
```
Define a fragment with a TextView inside its layout and add the attribute `android:letterSpacing="@dimen/custom_letter_spacing"` to the TextView
Add a dimen to `dimens.xml`:
```
<dimen name="custom_letter_spacing">0.2</dimen>
```
Create a test that uses the provided matcher on the fragment with fragment-testing inside a FragmentScenario.
`onView(withId(R.id.text_view_id)).check(matches(hasLetterSpacing(R.dimen.custom_letter_spacing)))
`

Robolectric is not able to set the new letterSpacing attribute resulting in a failing test

### Robolectric & Android Version
Roboelectric: 4.7.3
fragment-testing:1.4.1
targetSdk: 31
compileSdk: 31

### PR DESCRIPTION
**Pull request by bacecek**
### Overview
Added requested methods from the issue to `ShadowPaint` class.
### Proposed Changes
In additional to the new methods I added an Espresso test to check them where it's possible.
Also It seemed more consistent to me to use `textScaleX` parameter to measure the width of text (in method `measureText`), but here I'm not sure it's worth bothering to much.

Fixes #7246

### CODE REVIEW

File: integration_tests/androidx_test/src/main/res/layout/espresso_activity.xml [hunk 1]
```diff
@@ -23,6 +23,8 @@
       android:id="@+id/text_view"
       android:layout_height="wrap_content"
       android:layout_width="wrap_content"
+      android:letterSpacing="0.05"
+      android:textScaleX="1.5"
```
Comment (utzcoz) [inline integration_tests/androidx_test/src/main/res/layout/espresso_activity.xml:27]: What about adding another `TextView` with negative `textScaleX`?

File: integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java [hunk 2]
```diff
@@ -115,6 +116,13 @@ public void textView() {
               assertThat(view.getHeight()).isGreaterThan(0);
             });
     onView(withText("Text View")).check(matches(isCompletelyDisplayed()));
+    onView(withText("Text View"))
```
Comment (utzcoz) [inline integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java:119]: Maybe we can extract `Text View` to a string for this class.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowPaintTest.java [hunk 3]
```diff
@@ -78,6 +78,27 @@ public void measureTextActuallyMeasuresLength() {
     assertThat(paint.measureText(new StringBuilder("Hello"), 1, 4)).isEqualTo(3.0f);
   }
 
+  @Test
+  public void measureTextUsesTextScaleX() {
+    Paint paint = new Paint();
+    paint.setTextScaleX(1.5f);
+    assertThat(paint.measureText("Hello")).isEqualTo(7.5f);
+    assertThat(paint.measureText("Hello", 1, 3)).isEqualTo(3.0f);
```
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowPaintTest.java:86]: We can extract `Hello` to a constant string.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowPaintTest.java [hunk 4]
```diff
@@ -78,6 +78,27 @@ public void measureTextActuallyMeasuresLength() {
...
+    assertThat(paint.measureText(new StringBuilder("Hello"), 1, 4)).isEqualTo(4.5f);
+  }
+
+  @Test
+  public void textWidthWithNegativeScaleXIsTheSameAsPositive() {
+    Paint positive = new Paint();
+    Paint negative = new Paint();
+    positive.setTextScaleX(1.5f);
+    negative.setTextScaleX(-1.5f);
+    assertThat(positive.measureText("Hello")).isEqualTo(negative.measureText("Hello"));
```
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowPaintTest.java:96]: Maybe we can add a new assertion to test final `measureText` result, e.g.

```
assertThat(positive.measureText("Hello")).isEqualsTo(7.5f);
```

Comment (bacecek) [reply robolectric/src/test/java/org/robolectric/shadows/ShadowPaintTest.java:96]: It will duplicates another test in this class, isn't?

File: integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java (hunk 2 above)
Comment (bacecek) [reply integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java:119]: Yes, I thought about it, but I decided to follow the style in the whole class and do it consistently. But if you think it's better that way, no problem.

File: integration_tests/androidx_test/src/main/res/layout/espresso_activity.xml (hunk 1 above)
Comment (utzcoz) [reply integration_tests/androidx_test/src/main/res/layout/espresso_activity.xml:27]: Sorry, I don't notice it. We can keep it.

File: integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java (hunk 2 above)
Comment (utzcoz) [reply integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java:119]: Not a big deal. We can keep it for consistent.

File: integration_tests/androidx_test/src/main/res/layout/espresso_activity.xml (hunk 1 above)
Comment (utzcoz) [reply integration_tests/androidx_test/src/main/res/layout/espresso_activity.xml:27]: To test negative value of textScaleX. You have created test for negative testScaleX at  ShadowPaintTest, it's better if we can test it too with real resource mechanism .

Comment (bacecek) [reply integration_tests/androidx_test/src/main/res/layout/espresso_activity.xml:27]: Seems like I was wrong and described behaviour with the same width but other direction works only for `Button` but not for usual `TextView`.
I will probably delete it and use 0 for negative `textScaleX` values for measuring text.

Comment (bacecek) [reply integration_tests/androidx_test/src/main/res/layout/espresso_activity.xml:27]: Fixed.
Also added new Espresso test with `TextView` and negative `textScaleX`.

File: integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java [hunk 5]
```diff
@@ -115,6 +116,25 @@ public void textView() {
...
```
Comment (utzcoz) [inline integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java:119]: If you add a new test method called `textViewWithNegativeScaleX`, what about moving this part to a new method called `textViewWithPositiveScaleX`, and one called `testViewWithLetterSpacing`?

**Review by utzcoz — APPROVED** [review]

File: integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java (hunk 5 above)
Comment (bacecek) [reply integration_tests/androidx_test/src/test/java/org/robolectric/integrationtests/axt/EspressoTest.java]: Fixed.
I added one view for one attribute and one test for each view. Now the tests look more independent of each other.

**Comment by utzcoz** [comment]
And @bacecek , could you help to squash to one commit? Thanks.

**Review by hoisie — APPROVED** [review]
